spec/manifolds/cli_spec.rb

```diff
@@ -3,86 +3,102 @@
 require_relative "../../lib/manifolds/cli"
 require "fileutils"
 require "logger"
+require "fakefs/spec_helpers"
 
 RSpec.describe Manifolds::CLI do
+  include FakeFS::SpecHelpers
+
   let(:project_name) { "commerce" }
   let(:sub_project_name) { "Pages" }
   let(:null_logger) { Logger.new(File::NULL) }
-  let(:bq_service) { instance_double("Manifolds::Services::BigQueryService", "commerce") }
+
+  before do
+    FakeFS do
+      # Set up any template files that need to exist
+      FileUtils.mkdir_p("#{File.dirname(__FILE__)}/../../lib/manifolds/templates")
+      File.write("#{File.dirname(__FILE__)}/../../lib/manifolds/templates/config_template.yml", "vectors:\nmetrics:")
+      File.write("#{File.dirname(__FILE__)}/../../lib/manifolds/templates/vector_template.yml", "attributes:")
+    end
+  end
 
   describe "#init" do
     subject(:cli) { described_class.new(logger: null_logger) }
 
-    before do
-      allow(FileUtils).to receive(:mkdir_p)
-      allow(File).to receive(:open)
-      cli.init(project_name)
-    end
+    context "when initializing a new project" do
+      before { cli.init(project_name) }
 
-    it "creates the projects directory" do
-      expect(FileUtils).to have_received(:mkdir_p).with("./#{project_name}/projects")
+      it "creates a project directory structure" do
+        expect(Dir.exist?("./#{project_name}")).to be true
+        expect(Dir.exist?("./#{project_name}/projects")).to be true
+        expect(Dir.exist?("./#{project_name}/vectors")).to be true
+      end
     end
   end
 
   describe "#add" do
-    let(:cli) { described_class.new(logger: null_logger) }
+    subject(:cli) { described_class.new(logger: null_logger) }
 
-    context "when within an umbrella project" do
+    context "when adding a project within an umbrella project" do
       before do
-        FileUtils.mkdir_p("#{project_name}/projects") # Simulate an umbrella project
+        FileUtils.mkdir_p("#{project_name}/projects")
         Dir.chdir(project_name)
         cli.add(sub_project_name)
       end
 
       after do
         Dir.chdir("..")
-        FileUtils.rm_rf(project_name)
       end
 
-      it "creates a tables directory within the project" do
+      it "creates the expected project structure" do
         expect(Dir.exist?("./projects/#{sub_project_name}/tables")).to be true
-      end
-
-      it "creates a routines directory within the project" do
         expect(Dir.exist?("./projects/#{sub_project_name}/routines")).to be true
       end
```

Comment:
Again, split this into two tests or wrap the assertions in an `aggregate_failures` block.